Gems/LandscapeCanvas/Code/Source/LandscapeCanvasSystemComponent.cpp

```diff
@@ -82,29 +79,58 @@
 
 namespace LandscapeCanvas
 {
+    namespace Internal
+    {
+        // The FastNoise gem is optional, so we need to keep track of its component type ID
+        // ourselves since we can't rely on the headers being there.
+        static constexpr const char* EditorFastNoiseGradientComponentTypeId = "{FD018DE5-5EB4-4219-9D0C-CB3C55DE656B}";
+
+        // The Vegetation gem is optional, so we need to keep track of the component type IDs
+        // ourselves since we can't rely on the headers being there.
+        namespace Vegetation
+        {
+            static constexpr const char* EditorAreaBlenderComponentTypeId = "{374A5C69-A252-4C4B-AE10-A673EF7AFE82}";
+            static constexpr const char* EditorBlockerComponentTypeId = "{9E765835-9CEB-4AEC-A913-787D3D21451D}";
+            static constexpr const char* EditorMeshBlockerComponentTypeId = "{130F5DFF-EF6F-4B37-8717-194876DE12DB}";
```

Comment:
Any thoughts on whether it makes sense to make this data driven? perhaps a .setreg? Would be handy if a new shape/filter were added not to need to compile.

Reply:
(Well, we'd still need to compile the new shape or filter regardless...)

I'm 100% for this being data-driven, but I'm worried that moving these type IDs into a setreg will make them even harder to find and keep in sync than where they're at in code right now. We ultimately really need to find a way to only need a single copy of them instead of duplicating them.

Reply:
See comment below, would ideally make this data driven but will defer to a stretch goal.